### PR TITLE
[libffi] Update to 3.4.4. JB#56492

### PR DESCRIPTION
**Pull request by pvuorela**
- patch no longer needed
- disabled info page building as useless
- old .so copied to make sure the upgrade goes smooth

The cyclic dependency to itself is really something, but on quick test seems to work and the earlier .so copy cases must have had a similar thing, even if implicit. 

@mlehtima @mkosola

### CODE REVIEW

**Comment by mlehtima** [comment]
Submodule should be changed to point to mirror.

**Comment by Thaodan** [comment]
Was there an so version bump? Usually copying the old library isn't needed unless that is the case.

File: rpm/libffi.spec [hunk 1]
```diff
@@ -58,39 +58,31 @@ developing applications that use %{name}.
...
 %make_build
 
 %install
-rm -rf %{buildroot}
 %make_install
 
-rm -f $RPM_BUILD_ROOT%{_infodir}/dir
+# include old version to ensure smooth upgrade. to be removed later.
+cp -a /%{_libdir}/libffi.so.6 $RPM_BUILD_ROOT/%{_libdir}/
+cp -a /%{_libdir}/libffi.so.6.0.4 $RPM_BUILD_ROOT/%{_libdir}/
```
Comment (Thaodan) [inline rpm/libffi.spec]: The slash here is redundant.
See `rpm --eval '%{_libdir}'`

**Comment by pvuorela** [comment]
> Was there an so version bump?

Yes

**Review by mlehtima — APPROVED** [review]

**Review by mkosola — APPROVED** [review]

**Comment by Thaodan** [comment]
> > Was there an so version bump?
> 
> Yes

Ah I still wonder if we need to keep the old lib as it isn't used in the build chain, before everything worked without.
But better save then sorry.

**Review by Thaodan — APPROVED** [review]

**Pull request merged**